src/west/configuration.py

```diff
@@ -296,28 +314,43 @@ def set(self, option: str, value: Any, configfile: ConfigFile = ConfigFile.LOCAL
         :param configfile: type of config file to set the value in
         '''
 
+        def check_configfile(location: ConfigFile) -> Path:
+            '''
+            check that exactly one configfile is in use (even if it not exists yet).
+            Return its path.
+            '''
+            configs = self.get_paths(location)
+            if not configs:
+                raise WestNotFound(f'{configfile}: Cannot determine any config file')
+            if len(configs) > 1:
+                raise ValueError(f'Cannot set value if multiple configs in use: {configs}')
```

Comment:
This exception is raised in multiple places: raise from a single "check_single" function.